### PR TITLE
[CodeCamp] #50 Add module combination doc

### PR DESCRIPTION
**Pull request by Seperendity**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# 一、应用不同的Neck和Head
+
+1. 假设想将`YOLOv6RepPAFPN`作为`YOLOv5`的`Neck`，则配置文件如下：
+
+   ```markdown
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:5]:
```suggestion
   ```python
```

Comment (PeterH0323) [reply docs/zh_cn/advanced_guides/module_combination.md:5]: 其他也请做下改动

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
+# 一、应用不同的Neck和Head
```
Comment (PeterH0323) [inline docs/zh_cn/advanced_guides/module_combination.md:1]:
```suggestion
# 一、应用不同的 Neck 和 Head
```

Comment (PeterH0323) [reply docs/zh_cn/advanced_guides/module_combination.md:1]: 中文与英文之间需要一个空格

Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:1]: 你这个任务应该是包括： backbone+neck+headmodule 的组合替换。我的建议是：

以 YOLOv5 为核心主配置，分成3个小结

## 1 YOLOv5 Backbone 替换
1. RTMDet backbone + yolov5 neck/head
2. YOLOv6 backbone + yolov5 neck/head
## 2 YOLOv5 Neck 替换
1. YOLOv5 Backbone+YOLOv6 neck+ YOLOv5 head
2. YOLOv5 Backbone+RTMDet neck+ YOLOv5 head

## 3 YOLOv5 headmodule 替换
1. YOLOv5 Backbone+YOLOv5 neck+ YOLOv6 headmodule
2. YOLOv5 Backbone+YOLOv5 neck+ YOLOX headmodule

以上修改要确保运行不报错

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    neck = dict(
+        type = 'YOLOv6RepPAFPN',
+        in_channels = [256, 512, 1024],
+        out_channels = [128, 256, 512], # 注意YOLOv6RepPAFPN的输出通道是[128, 256, 512]
+        num_csp_blocks = 12,
+        act_cfg = dict(type='ReLU', inplace = True),
+    ),
+    bbox_head = dict(
+        head_module = dict(
+            in_channels = [128, 256, 512])) # head部分输入通道要做相应更改
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:57]:
```suggestion
            in_channels = [128, 256, 512])) # head 部分输入通道要做相应更改
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        widen_factor = widen_factor,
+        upsample_feats_cat_first = False,
+        in_channels = [256, 512, 1024],
+        out_channels = [128, 256, 512],
+        norm_cfg=dict(type='BN', momentum=0.03, eps=0.001),
+        act_cfg = dict(type='SiLU', inplace=True),
+    ),
+    bbox_head = dict(
+        head_module = dict(
+            in_channels = [256, 512, 1024])) # 注意使用YOLOv7PAFPN后head部分输入通道数是neck输出通道数的两倍
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:83]:
```suggestion
            in_channels = [256, 512, 1024])) # 注意使用 YOLOv7PAFPN 后 head 部分输入通道数是 neck 输出通道数的两倍
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 5]
```diff
@@ -0,0 +1,142 @@
+# Backbone + Neck + Head 的组合替换
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:1]:
```suggestion
# Backbone + Neck + HeadModule 的组合替换
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+        norm_cfg=dict(type='BN', momentum=0.03, eps=0.001),
+        act_cfg = dict(type='SiLU', inplace=True),
+    ),
+    bbox_head = dict(
+        head_module = dict(
+            in_channels = [256, 512, 1024])) # 注意使用YOLOv7PAFPN后head部分输入通道数是neck输出通道数的两倍
+)
+```
+
+## 3. YOLOv5 Head 替换
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:87]:
```suggestion
## 3. YOLOv5 HeadModule 替换
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+(1) 假设想将 `yolov5 backbone + yolov5 neck + yolo7 head` 作为 `YOLOv5` 的完整网络，则配置文件如下：
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:89]:
```suggestion
(1) 假设想将 `yolov5 backbone + yolov5 neck + yolo7 head module` 作为 `YOLOv5` 的完整网络，则配置文件如下：
```

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 8]
```diff
@@ -0,0 +1,142 @@
...
+model = dict(
+    bbox_head=dict(
+        type='YOLOv7Head',
+        head_module=dict(
+            type='YOLOv7HeadModule',
+            num_classes=num_classes,
+            in_channels=[256, 512, 1024],
+            featmap_strides=strides,
+            num_base_priors=3),
+        prior_generator=dict(
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:114]: 这些都不需要改。你只需要换 head module

File: yolov5_s-v61_syncbn_fast_8xb16-300e_coco.py [hunk 9]
```diff
@@ -0,0 +1,336 @@
+default_scope = 'mmyolo'
```
Comment (hhaAndroid) [inline yolov5_s-v61_syncbn_fast_8xb16-300e_coco.py:1]: 这个文件删掉，demo.jpg 文件也删掉

File: docs/zh_cn/advanced_guides/module_combination.md (hunk 8 above)
Comment (Seperendity) [reply docs/zh_cn/advanced_guides/module_combination.md:114]: 这里是只更换了 `head module` ，是需要我做其他什么修改嘛。您这里指什么不需要修改呢？

File: docs/zh_cn/advanced_guides/module_combination.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+```
+
+(2)  `YOLOv6EfficientRep backbone + yolov5 neck + yolov5 head` 作为 `YOLOv5` 的完整网络，则配置文件如下：
+
+```python
+_base_ = './yolov5_s-v61_syncbn_8xb16-300e_coco.py'
+
+model = dict(
+    backbone=dict(
+        type='YOLOv6EfficientRep',
```
Comment (hhaAndroid) [inline docs/zh_cn/advanced_guides/module_combination.md:34]: 是否需要 _delete_=True,？

Comment (Seperendity) [reply docs/zh_cn/advanced_guides/module_combination.md:34]: 应该不需要，我看了两者主干所需参数完全一致

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**